### ISSUE TITLE
Possibility to hide tabs in the left tabbar

### ISSUE THREAD

**Issue by EstherPerelman** [eclipse-theia/theia#9800]
### Feature Description:
The possibility to hide all tabs except of specific tabs by giving their views ids,
For example: given the next array:  ['explorer', 'dockerView'] should result the next (circled in red):
![image](https://user-images.githubusercontent.com/71729183/127474077-48225750-cf8b-4f9f-8f20-458d8022f97b.png)

**Comment by EstherPerelman** [comment]
Hi, How should I implement this feature - where to enable to set it?
We thought of 2 options: 
1. By adding a setting to preferences,
2. By adding a function to rpc which will be called by an extension,
WDYS?
cc @vince-fugnitto  @colin-grant-work

**Comment by vince-fugnitto** [comment]
@EstherPerelman I'm not sure I follow the feature, can you elaborate?
At the moment the framework supports specifying in which area views would be opened by default, do you somehow want a contribution or API that allows multiple views at once to be added to a particular area?

**Comment by EstherPerelman** [comment]
@vince-fugnitto 
In our product we want to show the user in the left sidebar only specific views (e.g: a custom view we contribute) but not the debug, explorer, scm and etc. (The user will still be able to open these views by the `open view` command or other ways)
In summary - we want to be able to control which icons will be shown in the sidebar, 
For that I would like to contribute theia a PR that will support to set which tabs to show...,
My question is: In our product - how will we get to set this option:
Should I create a new setting in preferences in my PR so we could set it in package.json in our product
or I'll add an api in the rpc in my PR and we will add an extension that uses it in our product?

**Comment by vince-fugnitto** [comment]
@EstherPerelman I believe it should already be supported, and such customizations would be made directly in your product.
You can extend the default locations of views, and implement their `initializeLayout`.

**Comment by EstherPerelman** [comment]
@vince-fugnitto we are already override the `createDefaultLayout` but we want to hide icons of `viewsContainers` contributions by other extensions (not the built-in) for example - docker is installed but we want to hide its icon - we can't override its `initializeLayout`

**Comment by EstherPerelman** [comment]
@vince-fugnitto WDYS about contributing a PR which supports the config:
`  "theia": {
    "frontend": {
      "config": {
        "defaultSidebarIcons": ["explorer", "dockerView",..]
         ...
      }
    }
  },`
Thus we will be able to determine which icons to show on loading (even those who contributed)
cc @colin-grant-work

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto we are already override the `createDefaultLayout` but we want to hide icons of `viewsContainers` contributions by other extensions (not the built-in) for example - `docker` extension is installed but we want to hide its icon - we can't override its `initializeLayout`

@EstherPerelman how would you expect to maintain a list of all plugins you want to hide from the sidebar? Plugins can be installed dynamically, including them in some config in the app's `package.json` will not scale well.

**Comment by EstherPerelman** [comment]
> @EstherPerelman how would you expect to maintain a list of all plugins you want to hide from the sidebar? Plugins can be installed dynamically, including them in some config in the app's `package.json` will not scale well.

@vince-fugnitto the list will include the plugins I want to **show** (not to hide), I have already found the [place where the icons are set from the list of the contributed `viewContainers`](https://github.com/eclipse-theia/theia/blob/5284996e010ddf5261d603d29ed0a1d83175a9be/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L535) I mean to check there if the containerViewId is exist in the `defaultSidebarIcons` array (already test it and worked),
Am I missing something?

**Comment by vince-fugnitto** [comment]
> the list will include the plugins I want to show (not to hide)

I'm not sure I follow, keeping a list of plugins to `show` would not scale well in my opinion.
Given a user installs 10 arbitrary plugins that each contribute view-containers, how will you manage which ones to actually show?

Alternatively, you can rebind `PluginViewRegistry` in your application to include the logic of determining what to hide/show, I don't think the extra config in the `package.json` for a specific use-case is necessary?

**Comment by idoprz** [comment]
> > the list will include the plugins I want to show (not to hide)
> 
> I'm not sure I follow, keeping a list of plugins to `show` would not scale well in my opinion.
> Given a user installs 10 arbitrary plugins that each contribute view-containers, how will you manage which ones to actually show?
> 
> Alternatively, you can rebind `PluginViewRegistry` in your application to include the logic of determining what to hide/show, I don't think the extra config in the `package.json` for a specific use-case is necessary?

We pre-install several plugins, but we do not want to clutter his activity view with too many icons on first time load. If he want to see them, he can go to view and select which one to show

The suggestion is to create a list of plugins we will show by default to the user. All other installed plugins will be closed.

**Comment by vince-fugnitto** [comment]
> We pre-install several plugins, but we do not want to clutter his activity view with too many icons on first time load. If he want to see them, he can go to view and select which one to show

Thank you for clarifying, the idea is to control the default layout with bundled plugins when a user starts a workspace for the first time.

> The suggestion is to create a list of plugins we will show by default to the user. All other installed plugins will be closed.

The config in the `package.json` is likely not necessary, it should already be supported if you rebind the method in your application and control the list in the code itself (like other framework customizations). The default behavior for end-users would be to display the view as intended, if your use-case intends to hide the view from the default layout it can be implemented that way in your application?

**Comment by EstherPerelman** [comment]
> it should already be supported if you rebind the method in your application

@vince-fugnitto thanks for replying, Did you talk about rebinding [this method](https://github.com/eclipse-theia/theia/blob/5284996e010ddf5261d603d29ed0a1d83175a9be/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L511)? actually to override it?

**Comment by vince-fugnitto** [comment]
> > it should already be supported if you rebind the method in your application
> 
> @vince-fugnitto thanks for replying, Did you talk about rebinding [this method](https://github.com/eclipse-theia/theia/blob/5284996e010ddf5261d603d29ed0a1d83175a9be/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L511)? actually to override it?

Right, rebinding `PluginViewRegistry` to override any method as necessary including `initWidgets`.
I assume for your product you can keep a list of extension ids you want to handle, and not display them directly in the code. For such a particular use-case I don't think including additional application config properties is necessary.

**Comment by EstherPerelman** [comment]
> Right, rebinding `PluginViewRegistry` to override any method as necessary including `initWidgets`.

@vince-fugnitto Actually we can do it in our side with dirty code because there is private objects and this duplicated code that need to be maintenance it every upgrade,
Thus I opened #9847 to make it easier to maintain, 
Can you please approve that?
